HW12/homework_12.py

```diff
@@ -0,0 +1,97 @@
+import json
+import time
+from datetime import datetime
+# coding=utf8
+
+
+def name_time_decorator(func):
+    def inner1(*args, **kwargs):
+        ret = func(*args)
+        file = open("func_log.txt", "a+")
+        file.write(f"Function '{func.__name__}' started at {datetime.now()}\n")
+        file.close()
+        return ret
+    return inner1
+
+
+@name_time_decorator
+def read(filename):
+    with open(filename, 'r') as json_file:
+        return json.load(json_file)
+
+
+@name_time_decorator
+def write(data, filename):
+    with open(filename, 'w') as json_file:
+        json.dump(data, json_file)
+
+
+
+# contacts = {'mom': '+1111111', 'dad': '+2222222', 'sister': '+3333333', 'brother': '+4444444', 'uncle': '+5555555'}
+
+""" Here are commands which you can use in this app:
+stats - shows q-ty of contacts
+add - add a new contact (name and phone)
+delete - delete contact be name
+contacts_list - shows name of all
+show_name - shows contact phone number by name
+show_all - shows full dict
+"""
+while True:
+    contacts = read("contact_book.json")
```

Comment:
You can read and init `contacts` once before the loop. You are updating file after every change to your `contacts` dict so you can be sure you will have the same data structure every iteration

Reply:
Also, handle a situation if there is no file created - initialize an empty dict